pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/cliflags.go

```diff
@@ -56,8 +49,15 @@ func CLIFlags(kubernetesVersion *semver.Version, criName extensionsv1alpha1.CRIN
 			"--container-runtime=remote",
 			"--container-runtime-endpoint="+containerd.PathSocketEndpoint,
 		)
-	} else if image != nil {
-		flags = append(flags, "--pod-infra-container-image="+image.String())
+	} else if criName == extensionsv1alpha1.CRINameDocker {
+		flags = append(flags,
+			"--network-plugin=cni",
+			"--cni-bin-dir=/opt/cni/bin/",
+			"--cni-conf-dir=/etc/cni/net.d/",
+			fmt.Sprintf("--image-pull-progress-deadline=%s", cliFlags.ImagePullProgressDeadline.Duration.String()))
```

Comment:
This flag is configurable in the `Shoot`: https://github.com/gardener/gardener/blob/7145aa25f9523945ec7bfbd257ec9101176ade4c/example/90-shoot.yaml#L77

Can you please add some validation that prevents it from being configured if only `containerd` is used?

Reply:
done.